src/Robo/Plugin/Traits/ResultCheckTrait.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Usher\Robo\Plugin\Traits;
+
+use Robo\Result;
+
+/**
+ * Trait to provide result checking functionality.
+ */
+trait ResultCheckTrait
+{
+    /**
+     * Check if accumulated Robo tasks within a Result were all successful.
+     *
+     * @param \Robo\Result $result
+     *   The primary result to check within.
+     *
+     * @return bool
+     *   Boolean indicating if all tasks within a result were successful.
+     *
+     * @throws \Robo\Exception\TaskException
+     */
+    public function resultTasksSuccessful(Result $result): bool
```

Comment:
We used to use this before we parallelized tasks. When you run `$result->accumulate()` it requires something like this to really check the result of each individual result within the result. I don't know why Robo doesn't offer a method to do this already. Perhaps this could be a contrib opportunity.

Reply:
Evidently this approach doesn't play nice with parallel task results so I removed it.